features/zones/Create.feature

```diff
@@ -71,3 +73,22 @@ Feature: Zones: Zone create flow
               status: {}
       """
     Then the "$zone-connected-scanner" element contains "The Zone “test” is now connected"
+
+  Scenario: The form shows expected error for <StatusCode> response
+    When the URL "/provision-zone" responds with
+      """
+      headers:
+        Status-Code: '<StatusCode>'
+      """
+    And I "type" "test" into the "$name-input" element
+    And I click the "$create-zone-button" element
+
+    Then the "$create-zone-button" element <CreateButtonDisabledState> disabled
+    And the "$create-zone-error" element contains "<ErrorTitle>"
+    And the "$connect-zone-instructions" element doesn't exist
+
+    Examples:
+      | StatusCode | CreateButtonDisabledState | ErrorTitle                                     |
+      | 400        | isn't                     | The Zone name test is invalid                  |
+      | 409        | is                        | A Zone with the name test already exists       |
+      | 500        | isn't                     | An error occurred while creating the Zone test |
```

Comment:
> In two minds whether to say something here as I get what you are doing, but! These cucumber tests are supposed to be super readable so you can easily see at a glance what it being tested. I wonder if we can make it slightly better by using enabled/disabled in the examples rather than is/isn't?

I’ve changed this around a bit so the examples include the attribute.

Reply:
> thinking also wondering what happens if test already exists, then I delete test in another tab (or someone else does from afar, or via the CLI or whatever). I then can't save it from here as the button is now disabled, is that right? If so, I'm wondering if this is a case of us trying to do too much?

I don’t think I’m following.

If you already have a Zone "test", trying to create it again will result in the 409 error case. If however the Zone "test" was deleted, it won’t (assuming the process in the backend cleaned up anything related and actually deleted the Zone).

If you try and create a Zone "test" that already exists and then delete it, then the button in the form will remain disabled.

So I’m not sure where the "too much" part that we’re possibly doing is?

Reply:
So if you have two people separately in different places, or even two different tabs. If I try to add `test` in one tab and it already exists, the button will be made disabled. If I delete `test` in the other tab, and come back to my original tab, the button is still disabled (currently I think?) even though `test` no longer exists. I have to refresh, or nav back and forth to get my button to be enabled again?

I know the tab example is a bit weird, but replace tab for "browser on a different machine" or just "user" and maybe that helps to explain what I'm thinking about.

Reply:
> So if you have two people separately in different places, or even two different tabs. If I try to add test in one tab and it already exists, the button will be made disabled. If I delete test in the other tab, and come back to my original tab, the button is still disabled (currently I think?) even though test no longer exists. I have to refresh, or nav back and forth to get my button to be enabled again?

Correct. And I think that's the correct behavior. The button suddenly becoming enabled would be very disorienting and in this example you either don’t know that the Zone was deleted and won’t expect it becoming available or you do know and then you will likely guess reloading and trying again will work.

Reply:
(Not sure why the changed assertions fail now 🤔)